src/Native/System.Native/pal_networking.cpp

```diff
@@ -392,3 +401,84 @@ extern "C" int32_t GetHostName(uint8_t* name, int32_t nameLength)
     size_t unsignedSize = UnsignedCast(nameLength);
     return gethostname(reinterpret_cast<char*>(name), unsignedSize);
 }
+
+extern "C" void EnumerateInterfaceAddresses(IPv4AddressFound onIpv4Found,
+                                            IPv6AddressFound onIpv6Found,
+                                            LinkLayerAddressFound onLinkLayerFound)
+{
+    ifaddrs *headAddr, *current;
+    if (getifaddrs(&headAddr) == -1)
+    {
+        assert(false && "getifaddrs failed.");
```

Comment:
This assert will be removed on Release and we will attempt to free the unalloc'd headAddr after the for loop. Should there be a return / error code here?